velox/exec/HashPartitionFunction.h

```diff
@@ -21,6 +21,12 @@
 
 namespace facebook::velox::exec {
 
+/// Calculates partition number for each row of the specified vector using a
+/// hash function. The constructor with hashBitRange parameter requires both
+/// hashBitRange and keyChannels to be non-empty. The constructor with
+/// numPartitions allows the keyChannels argument to be empty. If keyChannels is
```

Comment:
Can we enforce that numPartitions is one if keyChannels are empty? Or shall we have a special constructor for this case. And let Prestissimo to choose the right constructor? Thanks!

Reply:
> Can we enforce that numPartitions is one if keyChannels are empty?

No, this change is originally aimed to be a fix for the buggy plan, setting `numPartitions` to 1 won't fix it. Please see https://github.com/prestodb/presto/pull/20381.

> Or shall we have a special constructor for this case. And let Prestissimo to choose the right constructor?

I guess we'd better fix the planner, after which we won't need a special ctor.
Thanks.